app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala

```diff
@@ -37,25 +39,25 @@ class TaxCodeChangeService @Inject()(taxCodeChangeConnector: TaxCodeChangeConnec
       case _                                                           => throw new RuntimeException(s"Could not fetch tax code change")
     }
 
-  def hasTaxCodeChanged(nino: Nino)(implicit hc: HeaderCarrier): Future[HasTaxCodeChanged] = {
-    val hasTaxCodeChangedFuture: Future[Boolean] = taxCodeChanged(nino)
-    val taxCodeMismatchFuture: Future[TaiResponse] = taxCodeMismatch(nino)
+  def hasTaxCodeChanged(nino: Nino)(implicit hc: HeaderCarrier): Future[Either[TaxCodeError, HasTaxCodeChanged]] = {
+    val hasTaxCodeChangedFuture = taxCodeChanged(nino)
+    val taxCodeMismatchFuture = taxCodeMismatch(nino)
 
-    for {
-      hasTaxCodeChanged <- hasTaxCodeChangedFuture
-      taxCodeMismatch   <- taxCodeMismatchFuture
-    } yield {
-      (hasTaxCodeChanged, taxCodeMismatch) match {
-        case (_: Boolean, TaiSuccessResponseWithPayload(taxCodeMismatch: TaxCodeMismatch)) => {
+    val hasTaxChanged = for {
+      hasTaxCodeChanged <- EitherT(hasTaxCodeChangedFuture)
+      taxCodeMismatch   <- EitherT.right[TaxCodeError](taxCodeMismatchFuture)
```

Comment:
Never used cats before so will need an intro into it

Reply:
if u want i can give an intro to cats one of these days for the whole team